src/util-decode-mime.h

```diff
@@ -197,10 +199,9 @@ typedef struct MimeDecParseState {
     uint8_t bvremain[B64_BLOCK];  /**< Remainder from base64-decoded line */
     uint8_t bvr_len;  /**< Length of remainder from base64-decoded line */
     uint8_t data_chunk[DATA_CHUNK_SIZE];  /**< Buffer holding data chunk */
-#ifdef HAVE_NSS
-    HASHContext *md5_ctx;
-    uint8_t md5[MD5_LENGTH];
-#endif
+    SCMd5 *md5_ctx;
+    char md5hex[SC_MD5_HEX_LEN + 1];
```

Comment:
I'm not really happy with growing the data structure like this. Can we do it some other way?

Reply:
This is simplest.  We could:
- Go back to how it was before and just keep the binary hash.  Use snprintf while logging.
- FinalizeToHex in logging.  But this would allow the logger to run once.  So I'd have to add a "Clone" method to clone the context in the logger, then that could finalize to hex. This postpones the hex til logging, and still allows multiple loggers, at the expense of copying the context. This would shrink the struct as no buffer would be required at all.
Anything thing else would kind of be out of scope for this PR I think.

Reply:
Perhaps for now revert to the binary hash + snprintf. Seems less expensive than the Clone method.

Reply:
Ok, this has been reverted back to the snprintf loop. The relevant commit only changes the subject hashing now.